packages/client/src/features/tools/node-creation/node-creation-tool.ts

```diff
@@ -158,6 +162,25 @@ export class NodeCreationToolMouseListener extends DragAwareMouseListener {
         return result;
     }
 
+    protected getCreateNodeOperationArgs(insert: TrackedInsert, ctx: GModelElement, event: MouseEvent): Args | undefined {
+        let args = { ...this.triggerAction.args };
+        if (this.tool.changeBoundsManager.snapper) {
+            args = { ...args, snapper: this.tool.changeBoundsManager.snapper.constructor.name };
```

Comment:
I don't know the intended purpose for the snapper name argument, but the current approach maybe flawed for a couple of reasons:
- changeBoundsManger.snapper is optional so it can be undefined. In that case wouldn't throw `this.tool.changeBoundsManager.snapper.constructor.name` an error?
- The snapper does not necessarily have a constructor function (i.e. is a class). What would happen if the snapper is e.g. bound to a direct object (via toConstantValue?).
- contructor.name is unreliable. In production builds contructor names get minified. So it can/should not be used a stable identifier

Maybe we should introduce an optional id property in the `ISnapper` interface and use that instead?

Reply:
For the optional behavior, that is why we have the `if` before but I agree that getting the name like that may not be the best option. I think I may have overshot with this a bit and I'll remove it again due to the unreliable behavior that you mentioned.